sources/api/migration/migrator/src/main.rs

```diff
@@ -116,27 +121,37 @@ where
 }
 
 pub(crate) async fn run(args: &Args) -> Result<()> {
-    let migrated_datastore = perform_migrations(args).await?;
+    let current_version = get_current_version(&args.datastore_path).await?;
+
+    exit_unless_version_change(&current_version, args).await?;
+
+    let migrated_datastore = perform_migrations(&current_version, args).await?;
 
     // Remove all the weak setting and all metadata
     let datastore =
-        remove_weak_settings_and_metadata(migrated_datastore, &args.migrate_to_version).await?;
+        remove_transient_datastore_entries(migrated_datastore, &args.migrate_to_version).await?;
     flip_to_new_version(&args.migrate_to_version, datastore).await?;
 
     Ok(())
 }
 
-pub(crate) async fn perform_migrations(args: &Args) -> Result<PathBuf> {
-    // Get the directory we're working in.
-    let datastore_dir = args
-        .datastore_path
-        .parent()
-        .context(error::DataStoreLinkToRootSnafu {
-            path: &args.datastore_path,
-        })?;
+pub(crate) async fn exit_unless_version_change(
+    current_version: &Version,
+    args: &Args,
+) -> Result<()> {
+    if current_version == &args.migrate_to_version {
+        info!(
+            "Requested version {} matches version of given datastore at '{}'; nothing to do. Exiting from migrator.",
+            args.migrate_to_version,
+            args.datastore_path.display()
+        );
+        process::exit(0);
+    }
```

Comment:
This same check is duplicated in `perform_migrations`, which runs right after `exit_unless_version_change` is called. Do we need both?

Reply:
Technically no, but we need to compute `Direction`, at which point we need to handle the `None`.

I refactored it a bit to compute `Direction` at the same time that we make the decision whether or not to exit. It cost a bit in readability (I really liked the `exit_unless()` function for that reason) but I think it's still pretty clear from looking at `run()` that the migrator will flatly exit if there's no version change.